src/borg/archiver.py

```diff
@@ -527,7 +529,8 @@ def _process(self, archive, cache, matcher, exclude_caches, exclude_if_present,
                                               read_special=read_special, dry_run=dry_run)
                         return
                 if not dry_run:
-                    status = archive.process_dir(path, st)
+                    if not recurse_excluded_dir:
+                        status = archive.process_dir(path, st)
```

Comment:
Hm, does it matter if the intermediate dirs are archived before or after the excluded dir?  It could be possible to add them (at the time an excluded dir is archived) to a list of paths to be archived later.